Userland/Services/WindowServer/Menu.cpp

```diff
@@ -456,11 +456,64 @@ void Menu::clear_hovered_item()
     redraw();
 }
 
+void Menu::start_activation_animation(MenuItem& item)
+{
+    VERIFY(menu_window());
+    VERIFY(menu_window()->backing_store());
+    auto window = Window::construct(*this, WindowType::Menu);
+    window->set_frameless(true);
+    window->disable_hit_tests(true);
+    window->set_opacity(0.8f); // start out transparent so we don't have to recompute occlusions
+    window->disable_hit_tests(1.0f);
+    window->set_rect(item.rect().translated(m_menu_window->rect().location()));
+    window->set_event_handler([](Core::Event&) {
+        // ignore all events
+    });
+
+    VERIFY(window->backing_store());
+    Gfx::Painter painter(*window->backing_store());
+    painter.blit({}, *menu_window()->backing_store(), item.rect(), 1.0f, false);
+    window->invalidate();
+
+    struct AnimationInfo {
+        RefPtr<Core::Timer> timer;
+        RefPtr<Window> window;
+        u8 step { 8 }; // Must be even number!
+
+        AnimationInfo(NonnullRefPtr<Window>&& window)
+            : window(move(window))
+        {
+        }
+    };
+    auto* animation = new AnimationInfo(move(window));
```

Comment:
Can we put this in an `OwnPtr` (or `NonnullOwnPtr`) instead? Naked new is so C++98..

Reply:
I tried that, and it ended up being very ugly. What happens is that when you capture a `NonnullOwnPtr` in the lambda, you end up not being able to use it without a `const_cast` or capturing an additional raw pointer and only using the captured object for lifetime purposes. I can change it back, but this looked a lot cleaner, though less elegant indeed.